code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,909 @@
+
+/*
+Warping extracts crossbreed
+put up a rune with bluespace effects, lots of those runes are fluff or act as a passive buff, others are just griefind tools */
+
+
+/obj/item/slimecross/warping
+	name = "warped extract"
+	desc = "It just won't stay in place."
+	icon_state = "warping"
+	effect = "warping"
+	colour = "grey"
+	///what runes will be drawn depending on the crossbreed color
+	var/obj/effect/warped_rune/runepath
+	/// the number of "charge" a bluespace crossbreed start with
+	var/warp_charge = 1
+	///max number of charge, might be different depending on the crossbreed
+	var/max_charge = 1
+	///time it takes to store the rune back into the crossbreed
+	var/storing_time = 15
+	///time it takes to draw the rune
+	var/drawing_time = 15
+
+
+/obj/effect/warped_rune
+	name = "warped rune"
+	desc = "An unstable rune born of the depths of bluespace"
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "greyspace_rune"
+	move_resist = INFINITY  //here to avoid the rune being moved since it only sets it's turf once when it's drawn. doesn't include admin fuckery.
+	anchored = TRUE
+	layer = MID_TURF_LAYER
+	resistance_flags = FIRE_PROOF
+	///is only used for bluespace crystal erasing as of now
+	var/storing_time = 5
+	///Nearly all runes needs to know which turf they are on
+	var/turf/rune_turf
+	///starting cooldown of the rune.
+	var/cooldown = 0
+	///duration of the cooldown for the rune only applies to certain runes
+	var/max_cooldown = 100
+
+
+///runes can also be deleted by bluespace crystals if the xenobiologist is fucking everyone up and you need a way to destroy the rune really fast.
+/obj/effect/warped_rune/attackby(obj/item/stack/used_item, mob/user)
```

Comment:
```suggestion
/obj/effect/warped_rune/attackby(obj/item/used_item, mob/user)
```
Keep the incoming var correctly typed. Type cast it if needed, but don't give it a type it might not necessarily have on the proc redefinition.